shared/common-adapters/avatar-line.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react'
+import {AvatarSize} from '../common-adapters/avatar'
+import * as Kb from '../common-adapters'
+import * as Styles from '../styles/index'
+
+type Props = {
+  usernames: Array<string>
+  maxShown: number
+  size: AvatarSize
+  layout: 'horizontal' | 'vertical'
+}
+
+// TODO: consider making `diagonal` to replace MultiAvatar, but that's hard.
+
+const AvatarLine = (props: Props) => {
+  const usernamesToShow = props.usernames.slice(0, props.maxShown)
+  const extra = props.usernames.length - usernamesToShow.length
+  const reverse = {horizontal: 'horizontalReverse', vertical: 'verticalReverse'}
+  const styles = styleMap[props.size][props.layout]
+  return (
+    <Kb.Box2 direction={reverse[props.layout]} style={styles.container}>
+      {!!extra && (
+        <Kb.Box2 direction={props.layout} alignItems="center" style={styles.overflowBox}>
+          <Kb.Text type={getTextSize(props.size)} style={styles.text}>
+            +{extra}
+          </Kb.Text>
+        </Kb.Box2>
+      )}
+      {usernamesToShow
+        .map(username => (
+          <Kb.Avatar
+            size={props.size}
+            username={username}
+            key={username}
+            borderColor="white"
+            style={styles.avatar}
+          />
+        ))
+        .reverse()}
+    </Kb.Box2>
+  )
+}
+
+const getTextSize = size => (size >= 48 ? 'BodySmallBold' : 'BodyTinyBold')
+
+const styleMap = [128, 96, 64, 48, 32, 24, 16, 12].reduce(function(styles, size) {
```

Comment:
should pull this array into its own thing like
`const avatarSizes: ReadOnlyArray<AvatarSize> = [128,96...`